### PR TITLE
feat(events): Use SnubaEvent if option is turned on

### PR DESCRIPTION
**Pull request by lynnagara**
Prevents 404s on various events endpoints caused by sampling events when Snuba is on

### CODE REVIEW

**Comment by lynnagara** [comment]
We probably have to do the same in a few more endpoints, I wonder if there's any better solution

File: src/sentry/api/endpoints/event_owners.py [hunk 1]
```diff
@@ -21,7 +22,11 @@ def get(self, request, project, event_id):
         :auth: required
         """
 
-        event = Event.objects.from_event_id(event_id, project.id)
+        use_snuba = options.get('snuba.events-queries.enabled')
+
+        event_cls = SnubaEvent if use_snuba else Event
+
+        event = event_cls.objects.from_event_id(event_id, project.id)
```
Comment (markstory) [inline src/sentry/api/endpoints/event_owners.py]: Would it be outlandish to make `Event.objects.from_event_id` do this logic internally? That might help you play fewer rounds of whac-a-mole.

**Comment by lynnagara** [comment]
Think I hit all the ones we need to update now

**Comment by JTCunning** [comment]
Removing myself from review, but giving a generic +1 and following along.

**Review by alex-hofsteede — APPROVED** [review]
Yeah this seems good. 

+0.5 to Mark's comment about having the EventManager logic do this SnubaEvent/Event switching internally. If it can be done that way easily without coupling the 2 event types too much, then I would be for it.

**Pull request merged**